### PR TITLE
feat: sb start stop conversation

### PR DESCRIPTION
**Pull request by trkhoi**
- [x] Api for new config `$sb set-start-stop <emoji-start> <emoji-stop> #general`
- [x] Normal starboard will have `reaction_type = "message"` in table `guild_config_repost_reaction` 
- [x] Start stop starboard will have `reaction_type = "conversation"` in table `guild_config_repost_reaction`
- [x] Handle case has config sb start stop

### CODE REVIEW

**Review by catngh — APPROVED** [review]

File: pkg/entities/configs.go [hunk 1]
```diff
@@ -465,14 +465,25 @@ func (e *Entity) ConfigRepostReaction(req request.ConfigRepostRequest) error {
 		Emoji:           req.Emoji,
 		Quantity:        req.Quantity,
 		RepostChannelID: req.RepostChannelID,
+		ReactionType:    "message",
```
Comment (nguyennh4522) [inline pkg/entities/configs.go:468]: `message` should be define in /consts

File: pkg/entities/configs.go [hunk 2]
```diff
@@ -465,14 +465,25 @@ func (e *Entity) ConfigRepostReaction(req request.ConfigRepostRequest) error {
...
+	})
+}
+
+func (e *Entity) CreateConfigRepostReactionStartStop(req request.ConfigRepostReactionStartStop) error {
+	return e.repo.GuildConfigRepostReaction.UpsertOne(model.GuildConfigRepostReaction{
+		GuildID:         req.GuildID,
+		EmojiStart:      req.EmojiStart,
+		EmojiStop:       req.EmojiStop,
+		RepostChannelID: req.RepostChannelID,
+		ReactionType:    "conversation",
```
Comment (nguyennh4522) [inline pkg/entities/configs.go:478]: `conversation` should be defined in /consts

File: pkg/repo/conversation_repost_histories/pg.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	return tx.Commit().Error
+}
+
+func (pg *pg) Update(model *model.ConversationRepostHistories) error {
+	return pg.db.Save(model).Error
```
Comment (nguyennh4522) [inline pkg/repo/conversation_repost_histories/pg.go:35]: `Save()` can create a new record and it doesn't match with the meaning of this function

File: pkg/repo/conversation_repost_histories/pg.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+func (pg *pg) GetByGuildAndChannel(guildID, channelID string) (*model.ConversationRepostHistories, error) {
+	var model model.ConversationRepostHistories
+	err := pg.db.Where("guild_id = ? AND origin_channel_id = ? and origin_start_message_id != '' and (origin_stop_message_id = '') IS NOT FALSE", guildID, channelID).First(&model).Error
```
Comment (nguyennh4522) [inline pkg/repo/conversation_repost_histories/pg.go:40]: ` IS NOT FALSE` look redundant

File: pkg/routes/v1.go [hunk 5]
```diff
@@ -147,6 +147,7 @@ func NewRoutes(r *gin.Engine, h *handler.Handler, cfg config.Config) {
 			repostReactionGroup.GET("/:guild_id", h.GetRepostReactionConfigs)
 			repostReactionGroup.POST("", h.ConfigRepostReaction)
 			repostReactionGroup.DELETE("", h.RemoveRepostReactionConfig)
+			repostReactionGroup.POST("/start-stop", h.CreateConfigRepostReactionStartStop)
```
Comment (nguyennh4522) [inline pkg/routes/v1.go:150]: `/start-stop`  seem not clean, can you separate the 2 APIs?

File: pkg/repo/conversation_repost_histories/pg.go (hunk 3 above)
Comment (trkhoi) [reply pkg/repo/conversation_repost_histories/pg.go]: change to Save() with Where("id")

File: pkg/routes/v1.go (hunk 5 above)
Comment (trkhoi) [reply pkg/routes/v1.go:150]: i think no need, this is used for admin and they dont want to add config 2 times
image

**Review by nguyennh4522 — APPROVED** [review]

**Pull request merged**